### DISCUSSION TITLE
Load Transformers class model weights in HuggingFace Bert

### DISCUSSION THREAD

**Discussion by anjali-chadha** [lucidrains/vit-pytorch#268] [Q&A]
Hello folks!

I have a model which was originally trained using vit-pytorch's [Transformer](https://github.com/lucidrains/vit-pytorch/blob/main/vit_pytorch/vit.py#L67-L80) implementation.
I am interested in loading this model weights into equivalent BERT implementation in transformers library

How can we achieve this model conversion? Is there any conversion script to map the model weights from vit-pytorch implementation to transformers' implementation

**Comment by Sir-Sloth-The-Lazy** [comment]
There's no official conversion script, but the mapping is straightforward because
both use the same Transformer architecture (QKV attention + FFN). The mismatch is
naming convention, not structure.
Here's the weight mapping logic:
```python
import torch
from vit_pytorch import ViT
from transformers import BertModel, BertConfig

vit = ViT(...)
vit.load_state_dict(torch.load("your_vit.pt"))

vit-pytorch naming → HuggingFace BERT naming (per layer)
vit: transformer.layers.{i}.0.norm.weight → bert: encoder.layer.{i}.attention.output.LayerNorm.weight
vit: transformer.layers.{i}.0.fn.to_qkv.weight → split into query/key/value weights
vit: transformer.layers.{i}.1.fn.net.0.weight → bert: encoder.layer.{i}.intermediate.dense.weight
state_dict = vit.state_dict()
new_state_dict = {}
for k, v in state_dict.items():
# map keys manually based on the above pattern
new_key = map_key(k) # your mapping function
new_state_dict[new_key] = v

The trickiest part: vit-pytorch's `to_qkv` is a single [dim, 3*dim] weight that must
be split into separate Q, K, V matrices for HuggingFace.
If you share the exact vit-pytorch config (dim, depth, heads), I can write the
complete mapping for you.
